src/app/analysis/analysis-detail/analysis-detail.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from '@angular/core';
+import {ActivatedRoute, Params, Router} from '@angular/router';
+import {ApiFileService} from '../../services/api-file.service';
+import {NgxSpinnerService} from 'ngx-spinner';
+import {Title} from '@angular/platform-browser';
+
+@Component({
+  selector: 'app-analysis-detail',
+  templateUrl: './analysis-detail.component.html',
+  styleUrls: ['./analysis-detail.component.css']
+})
+export class AnalysisDetailComponent implements OnInit {
+  accession: string;
+  analysis: any;
+  error: any;
+
+  constructor(private route: ActivatedRoute,
+              private router: Router,
+              private apiFileService: ApiFileService,
+              private spinner: NgxSpinnerService,
+              private titleService: Title) { }
+
+  ngOnInit() {
+    this.spinner.show();
+    this.route.params.subscribe((params: Params) => {
+      this.accession = params['id'];
+      this.titleService.setTitle(`${this.accession} | FAANG analysis`);
+    });
+    this.apiFileService.getAnalysis(this.accession).subscribe(
+      (data: any) => {
+        // if (data['hits']['hits'].length === 0) {
```

Comment:
Why this code is commented out? If you don't need it, please remove it completely, but we definitely need to have redirect to '404' page in case record doesn't exist.

Reply:
I have no knowledge of the document to tell me to add analysis index into API, therefore has to use the data directly from ES. Will change as long as the analysis data available from API

Reply:
please check commit https://github.com/FAANG/dcc-portal-frontend/pull/17/commits/71211ffd65f2a3d059ad74c05bf53f8d715113b3